partiql-eval/src/main/java/org/partiql/eval/value/DatumCollection.java

```diff
@@ -36,4 +36,34 @@ public Iterator<Datum> iterator() {
     public PType getType() {
         return _type;
     }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(_type);
+        sb.append("::");
+        if (_type.getKind() == PType.Kind.LIST) {
+            sb.append("[");
+        } else if (_type.getKind() == PType.Kind.BAG) {
+            sb.append("<<");
+        } else if (_type.getKind() == PType.Kind.SEXP) {
+            sb.append("(");
+        }
+        Iterator<Datum> iter = _value.iterator();
+        while (iter.hasNext()) {
+            Datum child = iter.next();
+            sb.append(child);
+            if (iter.hasNext()) {
+                sb.append(", ");
+            }
+        }
+        if (_type.getKind() == PType.Kind.LIST) {
+            sb.append("]");
+        } else if (_type.getKind() == PType.Kind.BAG) {
+            sb.append(">>");
+        } else if (_type.getKind() == PType.Kind.SEXP) {
+            sb.append(")");
+        }
+        return sb.toString();
+    }
```

Comment:
Interesting. Perhaps add a comment about being for debug?

Reply:
This is outside the scope of the PR, but I would suggest using the Ion (and PartiQLValue) reader/writer style and have toString() just invoke a text writer btw.